### PR TITLE
Restructure supported.rst into tutorial subpages

### LINKED ISSUE #623: Split "Supported software" into separate (more elaborate) pages
The [Supported Software](https://vdirsyncer.pimutils.org/en/stable/supported.html) page (`docs/supported.rst` in the git repo) contains too many sections. We should move almost all of these sections (certainly all sections that describe a server setup) to their own page in the [Other tutorials](https://vdirsyncer.pimutils.org/en/stable/tutorials/index.html) folder (`docs/tutorials/`) and add more details.

### PR DESCRIPTION
**Pull request by untitaker**
Fix #623

### CODE REVIEW

**Comment by polyzen** [comment]
[Reference labels](http://www.sphinx-doc.org/en/stable/markup/inline.html#role-ref) (such as ` .. _davmail_setup:`) are no longer useful as they are their own pages and should be referenced with the :doc: role.

Top sections should use `=`: http://www.sphinx-doc.org/en/stable/rest.html#sections

Irrelevant: probably want to make more use of external link targets.

File: docs/tutorials/owncloud.rst [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    username = ...
+    password = ...
+
+- *Versions older than 7.0.0:* ownCloud uses SabreDAV, which had problems
+  detecting collisions and race-conditions. The problems were reported and are
+  fixed in SabreDAV's repo, and the corresponding fix is also in ownCloud since
+  7.0.0. See :gh:`16` for more information.
+
+.. _ownCloud: https://owncloud.org/
+
```
Comment (polyzen) [inline docs/tutorials/owncloud.rst]: Excess whitespace.

File: docs/tutorials/todoman.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+The iCalendar format also supports saving tasks in form of ``VTODO``-entries,
+with the same file extension as normal events: ``.ics``. Many CalDAV servers
+support synchronizing tasks, vdirsyncer does too.
+
+todoman_ is a CLI task manager supporting :doc:`vdir <vdir>`. Its interface is
+similar to the ones of Taskwarrior or the todo.txt CLI app. You can use
+:storage:`filesystem` with it.
+
+
```
Comment (polyzen) [inline docs/tutorials/todoman.rst]: Excess whitespace.

**Comment by untitaker** [comment]
Thanks for your comments. I fixed the spacing and section headings, but would like to keep the refs in case I restructure again or just want to change the title of the page.

File: docs/tutorials/fastmail.rst [hunk 3]
```diff
@@ -0,0 +1,22 @@
+========
+FastMail
+========
+
+Vdirsyncer is irregularly tested against FastMail_. There are no known issues
```
Comment (hobarrera) [inline docs/tutorials/fastmail.rst]: Isn't it **regularly** tested against FastMail?

Comment (untitaker) [inline docs/tutorials/fastmail.rst]: yes, please file a PR. This is outdated

**Pull request merged**